xbmc/guilib/GUIVecText.h

```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright (C) 2005-2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "utils/ColorUtils.h"
+
+#include <stdint.h>
+#include <vector>
+
+struct character_t
+{
+  character_t(char l, int s, UTILS::COLOR::Color c)
+    : letter(static_cast<char32_t>(l)), style(s), color(c)
+  {
+  }
+  character_t(char16_t l, int s, UTILS::COLOR::Color c)
+    : letter(static_cast<char32_t>(l)), style(s), color(c)
+  {
+  }
+  character_t(wchar_t l, int s, UTILS::COLOR::Color c)
+    : letter(static_cast<char32_t>(l)), style(s), color(c)
+  {
+  }
+  character_t(char32_t l, int s, UTILS::COLOR::Color c) : letter(l), style(s), color(c) {}
+
```

Comment:
here in character_t struct there are mixed types
`UTILS::COLOR::ColorIndex` VS `UTILS::COLOR::Color`
is intentional?